src/backend/distributed/utils/node_metadata.c

```diff
@@ -940,3 +919,48 @@ TupleToWorkerNode(TupleDesc tupleDescriptor, HeapTuple heapTuple)
 
 	return workerNode;
 }
+
+
+/*
+ * ExtractDatumFromString transforms a string representation into a Datum.
+ */
+Datum
+ExtractDatumFromString(char *inputString, Oid dataType)
+{
+	Oid typIoFunc = InvalidOid;
+	Oid typIoParam = InvalidOid;
+	int32 typeModifier = -1;
+	Datum datum = 0;
+	FmgrInfo fmgrInfo;
+	memset(&fmgrInfo, 0, sizeof(fmgrInfo));
+
+	getTypeInputInfo(dataType, &typIoFunc, &typIoParam);
+	fmgr_info(typIoFunc, &fmgrInfo);
+
+	getBaseTypeAndTypmod(dataType, &typeModifier);
+
+	datum = InputFunctionCall(&fmgrInfo, inputString, typIoParam, typeModifier);
+
+	return datum;
+}
+
+
+/*
+ * ExtractStringFromDatum returns the string representation of the given datum.
+ */
+char *
+ExtractStringFromDatum(Datum datum, Oid dataType)
+{
+	char *outputString = NULL;
+	Oid typIoFunc = InvalidOid;
+	bool typIsVarlena = false;
+	FmgrInfo fmgrInfo;
+	memset(&fmgrInfo, 0, sizeof(fmgrInfo));
+
+	getTypeOutputInfo(dataType, &typIoFunc, &typIsVarlena);
+	fmgr_info(typIoFunc, &fmgrInfo);
+
+	outputString = OutputFunctionCall(&fmgrInfo, datum);
```

Comment:
You can probably use `OidOutputFunctionCall` to save yourself the call to `fmgr_info` here.